From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Akanksha Hotta <akanksha.hotta@intel.com>
Date: Mon, 31 Aug 2026 11:37:27 +0530
Subject: drm/xe: Backport pin_vram_percent module parameter

The reason of failure is the pin_vram_percent member of struct xe_modparam is
conditionally compiled under the macro BPM_DRMM_CGROUP_REGISTER_REGION_NOT_PRESENT.
However, in module param registeration it was earlier declared
unconditionally without guarding under the macro.

Signed-off-by: Akanksha Hotta <akanksha.hotta@intel.com>
---
 drivers/gpu/drm/xe/xe_module.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/xe/xe_module.c b/drivers/gpu/drm/xe/xe_module.c
index cc58ddd..bdeccc1 100644
--- a/drivers/gpu/drm/xe/xe_module.c
+++ b/drivers/gpu/drm/xe/xe_module.c
@@ -41,9 +41,11 @@ module_param_named(svm_notifier_size, xe_modparam.svm_notifier_size, uint, 0600)
 MODULE_PARM_DESC(svm_notifier_size, "Set the svm notifier size in MiB, must be power of 2 "
 		 "[default=" __stringify(XE_DEFAULT_SVM_NOTIFIER_SIZE) "]");
 
+#ifdef BPM_DRMM_CGROUP_REGISTER_REGION_NOT_PRESENT
 module_param_named(pin_vram_percent, xe_modparam.pin_vram_percent, uint, 0400);
 MODULE_PARM_DESC(pin_vram_percent, "Max percent of VRAM pinnable via dma-buf per region (0=unlimited) "
                 "[default=" __stringify(XE_DEFAULT_PIN_VRAM_PERCENT) "]");
+#endif
 
 module_param_named_unsafe(force_execlist, xe_modparam.force_execlist, bool, 0444);
 MODULE_PARM_DESC(force_execlist, "Force Execlist submission");
-- 
2.43.0

